Add likes function

yoh2140 opened the pull request:
# WHAT
いいね機能の実装（非同期通信）

# WHY
人気記事を、後の人気順に反映させる為

Commit:
add like model

Commit:
add column to like table,
t.references :protottype
t.references :user

Commit:
add associations belongs_to
:prototype counter_cache: :likes_count
:user

Commit:
add column to prototype table, likes_count:integer

Commit:
add associations to prototype. user,
has_many :likes

Commit:
add likes controller

Commit:
add likes routing, only create. destroy

Commit:
add render _like.html.haml for add likes folder

Commit:
column name spell miss, protottype_id → prototype_id, to like table

Commit:
add gem ‘pry-rails’

Commit:
add render likes/like,
prototype: @prototype, like: @like

Commit:
add like views, partial template,
if like it delete response,
if not like it create response,

Commit:
add likes create view, ajax processing

Commit:
add like destroy view, ajax processing,

Commit:
add find_by method to likes user.id

Commit:
add like instance variable to prototype show action, likes user_id associated with the prototype,

Commit:
add create action to likes_controller, Like.create user_id(current_user) and prototype_id(prototype.id),
add delete action to likes_controller, destroy Like.id, after that prototype.id to @prototype

kenchan0503 commented on Gemfile:
```diff
@@ -51,3 +51,4 @@ gem 'bootstrap-sass'
 gem 'devise'
 gem 'carrierwave'
 gem 'kaminari'
+gem 'pry-rails'
```
pry-railsは本番環境で使用しないのでグルーピングしましょう！

yoh2140 replied:
OKです

kenchan0503 commented on db/migrate/20170320105335_create_likes.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    create_table :likes do |t|
+      t.references :protottype
+      t.references :user
```
外部キー制約もお願いいたします

yoh2140 replied:
忘れていました…。変更時につまづいてしまい一度カラムを削除し、再追加するという処置で対応しています。

kenchan0503 replied:
👍

kenchan0503 commented:
いいねが昨日しているgifアニメーションもお願いいたします！

Commit:
update pry-rails in group development

Commit:
delete column prptotype_id user_id to likes

Commit:
create prototype_id user_id to like
prototype_id:  index: true, foreign_key: true
user_id:  index: true, foreign_key: true

Commit:
update Gemfile.lock

yoh2140 commented:
![like_function](https://cloud.githubusercontent.com/assets/19721458/24202178/ad3f6e48-0f55-11e7-80ce-e41d8cc80247.gif)

いいね機能のGIFです。

kenchan0503 commented:
いーねした際に画像が置き変わっていないですね。
本番ですと赤色になるはずなのでお願いいたします。

Commit:
update like icon, file name icon_heart_red.svg

yoh2140 commented:
![like_function](https://cloud.githubusercontent.com/assets/19721458/24244303/9eb42b1e-1001-11e7-8f45-72da3a7b46f4.gif)

出来ました！
